src/Symfony/Component/DependencyInjection/Compiler/RandomizePrivateServiceIdentifiers.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Alias;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Randomizes private service identifiers, effectively making them unaccessable from outside.
+ *
+ * @author Roland Franssen <franssen.roland@gmail.com>
+ */
+class RandomizePrivateServiceIdentifiers implements CompilerPassInterface
```

Comment:
Reconsidered :)

> Because it makes no sense to randomize with XmlDumper and in fact would be wrong: it would rename services for no reason and make dumps harder and not accurate.

My first thought was, yeah this totally counts for a `PhpDumper` as well. Which in some ways is still true, but i guess you're right. Randomizing could be _the way_ a php dumper deals with privates by design (whereas xml adds `public="false"` :+1: ).

Do we need to consider DX here?
- Make it depend on some debug flag?
- Have 2 implementations (think `DebugPhpDumper`)
- Forget about randomizing, and compile different method names (think `getPrivate(.+)Service`)